documentation/internal/resource-manager-active-directory.md

```diff
@@ -0,0 +1,272 @@
+    - Updates properties of an application based on the parameters provided (`PATCH` operation).
+
+From the [_Manage app and resource access using Azure Active Directory groups_](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-manage-groups#how-does-access-management-in-azure-ad-work) document, an Azure AD group does the following:
+
+_Using groups lets the resource owner (or Azure AD directory owner), assign a set of access permissions to all the members of the group, instead of having to provide the rights one-by-one. The resource or directory owner can also give management rights for the member list to someone else, such as a department manager or a Helpdesk administrator, letting that person add and remove members, as needed._
+
+The cmdlets currently found in the `Az.Resources` module that act upon groups are the following:
+
+- [`Get-AzADGroup`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Get-AzADGroup.md)
+    - Retrieves all groups in the current tenant. The user can also filter results by object ID and display name.
+- [`New-AzADGroup`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/New-AzADGroup.md)
+    - Creates a new group in the current tenant with the given properties.
+- [`Remove-AzADGroup`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Remove-AzADGroup.md)
+    - Deletes a group with the given object ID or display name.
+
+From the [_Application and service principal objects in Azure Active Directory_](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Test-AzResourceGroupDeployment.md) document, a service principal is defined as the following:
+
+_When an application is given permission to access resources in a tenant (upon registration or consent), a service principal object is created._
+
+_A service principal must be created in each tenant where the application is used, enabling it to establish an identity for sign-in and/or access to resources being secured by the tenant. A single-tenant application has only one service principal (in its home tenant), created and consented for use during application registration. A multi-tenant Web application/API also has a service principal created in each tenant where a user from that tenant has consented to its use._
+
+The cmdlets currently found in the `Az.Resources` module that act upon service principals are the following:
+
+- [`Get-AzADServicePrincipal`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Get-AzADServicePrincipal.md)
+     - Retrieves all service principals in the current tenant. The user can also filter results by object ID, application ID, display name and service principal name.
+- [`New-AzADServicePrincipal`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/New-AzADServicePrincipal.md)
```

Comment:
Please explain a bit more about the intended scenario - why do we have c amcldte that does all these things, how is it implemented, what are its dependecies, and what are any current known issues?

Reply:
@markcowl updated -- called out why we want to enable this scenario, how it works with the Authorization SDK, gave some examples, and left a note about what we can do in the future to reach parity with the CLI scenario.